### ISSUE TITLE
Bug when using --clean option

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/policybrain-builder#67]
I'm getting an error under Python 2.7 that I never got before.
There is no import statement for shutil in the `cli/util.py` file; is that the problem?
```
iMac2:policybrain-builder mrh$ python setup.py install
/Users/mrh/anaconda/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'package_dirs'
  warnings.warn(msg)
running install
running bdist_egg
running egg_info
writing requirements to policybrain_builder.egg-info/requires.txt
writing policybrain_builder.egg-info/PKG-INFO
writing top-level names to policybrain_builder.egg-info/top_level.txt
writing dependency_links to policybrain_builder.egg-info/dependency_links.txt
writing entry points to policybrain_builder.egg-info/entry_points.txt
reading manifest file 'policybrain_builder.egg-info/SOURCES.txt'
writing manifest file 'policybrain_builder.egg-info/SOURCES.txt'
installing library code to build/bdist.macosx-10.7-x86_64/egg
running install_lib
running build_py
creating build/bdist.macosx-10.7-x86_64/egg
creating build/bdist.macosx-10.7-x86_64/egg/policybrain_builder
copying build/lib/policybrain_builder/__init__.py -> build/bdist.macosx-10.7-x86_64/egg/policybrain_builder
creating build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli
copying build/lib/policybrain_builder/cli/__init__.py -> build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli
copying build/lib/policybrain_builder/cli/config.py -> build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli
copying build/lib/policybrain_builder/cli/main.py -> build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli
copying build/lib/policybrain_builder/cli/package.py -> build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli
copying build/lib/policybrain_builder/cli/repository.py -> build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli
copying build/lib/policybrain_builder/cli/utils.py -> build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli
byte-compiling build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/__init__.py to __init__.pyc
byte-compiling build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli/__init__.py to __init__.pyc
byte-compiling build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli/config.py to config.pyc
byte-compiling build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli/main.py to main.pyc
byte-compiling build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli/package.py to package.pyc
byte-compiling build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli/repository.py to repository.pyc
byte-compiling build/bdist.macosx-10.7-x86_64/egg/policybrain_builder/cli/utils.py to utils.pyc
creating build/bdist.macosx-10.7-x86_64/egg/EGG-INFO
copying policybrain_builder.egg-info/PKG-INFO -> build/bdist.macosx-10.7-x86_64/egg/EGG-INFO
copying policybrain_builder.egg-info/SOURCES.txt -> build/bdist.macosx-10.7-x86_64/egg/EGG-INFO
copying policybrain_builder.egg-info/dependency_links.txt -> build/bdist.macosx-10.7-x86_64/egg/EGG-INFO
copying policybrain_builder.egg-info/entry_points.txt -> build/bdist.macosx-10.7-x86_64/egg/EGG-INFO
copying policybrain_builder.egg-info/not-zip-safe -> build/bdist.macosx-10.7-x86_64/egg/EGG-INFO
copying policybrain_builder.egg-info/requires.txt -> build/bdist.macosx-10.7-x86_64/egg/EGG-INFO
copying policybrain_builder.egg-info/top_level.txt -> build/bdist.macosx-10.7-x86_64/egg/EGG-INFO
creating 'dist/policybrain_builder-0.0.1-py2.7.egg' and adding 'build/bdist.macosx-10.7-x86_64/egg' to it
removing 'build/bdist.macosx-10.7-x86_64/egg' (and everything under it)
Processing policybrain_builder-0.0.1-py2.7.egg
creating /Users/mrh/anaconda/lib/python2.7/site-packages/policybrain_builder-0.0.1-py2.7.egg
Extracting policybrain_builder-0.0.1-py2.7.egg to /Users/mrh/anaconda/lib/python2.7/site-packages
Adding policybrain-builder 0.0.1 to easy-install.pth file
Installing pb script to /Users/mrh/anaconda/bin
Installing policybrain script to /Users/mrh/anaconda/bin

Installed /Users/mrh/anaconda/lib/python2.7/site-packages/policybrain_builder-0.0.1-py2.7.egg
Processing dependencies for policybrain-builder==0.0.1
Searching for click==6.7
Best match: click 6.7
Adding click 6.7 to easy-install.pth file

Using /Users/mrh/anaconda/lib/python2.7/site-packages
Finished processing dependencies for policybrain-builder==0.0.1


iMac2:policybrain-builder mrh$ ./taxcalc-package-cli.sh 0.10.1
STARTING : Fri Sep  8 06:37:06 EDT 2017
INFO: saved previous conda config: always_yes
INFO: changed conda config: always_yes True
INFO: removing directory: /tmp
INFO: restored previous conda config: always_yes
Traceback (most recent call last):
  File "/Users/mrh/anaconda/lib/python2.7/site-packages/policybrain_builder-0.0.1-py2.7.egg/policybrain_builder/cli/main.py", line 19, in start
    cli(obj={})
  File "/Users/mrh/anaconda/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/mrh/anaconda/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/mrh/anaconda/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/mrh/anaconda/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/mrh/anaconda/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/mrh/anaconda/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/mrh/anaconda/lib/python2.7/site-packages/policybrain_builder-0.0.1-py2.7.egg/policybrain_builder/cli/main.py", line 138, in release
    u.ensure_directory_exists(workdir, clean)
  File "/Users/mrh/anaconda/lib/python2.7/site-packages/policybrain_builder-0.0.1-py2.7.egg/policybrain_builder/cli/utils.py", line 82, in ensure_directory_exists
    shutil.rmtree(path)
NameError: global name 'shutil' is not defined

FINISHED : Fri Sep  8 06:37:09 EDT 2017
iMac2:policybrain-builder mrh$ 
```

**Comment by martinholmer** [comment]
@jbcrail,  When I add `import shutil` at the top of `cli/util.py` and then execute `python setup.py install` again, I get this error message:
```
.
.
.
    shutil.rmtree(path)
  File "/Users/mrh/anaconda/lib/python2.7/shutil.py", line 232, in rmtree
    onerror(os.path.islink, path, sys.exc_info())
  File "/Users/mrh/anaconda/lib/python2.7/shutil.py", line 230, in rmtree
    raise OSError("Cannot call rmtree on a symbolic link")
OSError: Cannot call rmtree on a symbolic link

```
What's going on?

I need to release bug-fix packages (for taxcalc 0.10.1) today, but can't get your CLI pb to work.
Any help would be much appreciated.

**Comment by martinholmer** [comment]
@jbcrail,  I'm curious about the CLI use of the shutil package.
I don't have that package installed on my Mac (which is running Python 2.7).
And I don't see it when doing `conda search shutil`.
How can this package be a dependency of policybrain-builder, when it seems so difficult to install?

**Comment by jbcrail** [comment]
@martinholmer `shutil` is part of the Python standard library and doesn't exist as a separate package.

Thanks for finding the missing import. This was part of the issue, but how we cleaned the working directory in the beginning was the main issue. By default, the working directory where we build the packages is "/tmp". Inside that directory, we create a "policybrain-builder" subdirectory containing the actual repos and packages.

Instead of cleaning "/tmp/policybrain-builder", we were cleaning "/tmp", which `shutil.rmtree` couldn't because it's typically owned by root. I updated the code to reflect this.
